src/json.hpp

```diff
@@ -2880,34 +2833,30 @@ class lexer
 
   private:
     /// input adapter
-    detail::input_adapter_t ia = nullptr;
+    detail::input_adapter_t ia { nullptr };
```

Comment:
I removed the `-Weffc++` warnings. Please use the `=`-style initializations again.

Reply:
I think using initializer-lists is really the future path of C++ member initialization; shouldn't we generally be going in that direction, instead?

Reply:
I agree that direct-list-initialization has some advantages here. No narrowing, and it considers explicit constructors as well.

I'm not a fan of member initialization when it's redundant (e.g. `std::string s{}`), but I would go for brace-style here too.

Reply:
I'm ok with letting it as it is. But just because braces work everywhere does not mean that `ia = nullptr;` is wrong in any sense. That's why *I* don't think we should change existing code.

Reply:
Right, it's not a refactoring/style PR anyway, I agree changing this code here is not that relevant.